### ISSUE TITLE
Padding (and even children Margin) clips children on Android

### ISSUE THREAD

**Issue by roubachof** [unoplatform/uno#12877]
### Current behavior

It seems that when setting Padding on a parent view, it clips children on android, whereas on other platform it does not clip the children.

### Expected behavior

Padding and margin should behave consistently between platform hence android shouldn't clip child when using padding or child margin.

### How to reproduce it (as minimally and precisely as possible)

Launch Uno toolkit winui sample, and select the ShadowContainer sample, from this branch:
https://github.com/unoplatform/uno.toolkit.ui/pull/571

![Screenshot_1689174238](https://github.com/unoplatform/uno/assets/596903/730817f7-ee1f-4249-b5af-9dbf833271b6)

you will see that the first "Neumorphism" card is clipped by its parent padding

```xml
   <StackPanel Width="400"
                        Margin="0,32"
                        Padding="32"
                        Background="{StaticResource UnoColor}"
                        CornerRadius="30">
                <ui:ShadowContainer Shadows="{StaticResource NeumorphismRaising}">
                    <Grid Width="300"
                          Padding="20"
                          Background="{StaticResource UnoColor}"
                          CornerRadius="20">
                        <Grid.RowDefinitions>
                            <RowDefinition Height="20" />
                            <RowDefinition Height="20" />
                        </Grid.RowDefinitions>

                        <TextBlock FontSize="15"
                                   Foreground="White"
                                   Text="Neumorphism" />
                        <TextBlock Grid.Row="1"
                                   FontSize="12"
                                   Foreground="White"
                                   Text="Raising element" />
                    </Grid>
                </ui:ShadowContainer>
```

on Windows and iOS, the children shadows are not clipped.
![windows_padding](https://github.com/unoplatform/uno/assets/596903/889af342-df1e-439a-beff-76156e7f64f6)
![ios_padding](https://github.com/unoplatform/uno/assets/596903/1ff649de-de24-4ab9-9078-89616d587216)

remark: Shadows are implement by a SKXamlCanvas .

### Works on UWP/WinUI

None

### NuGet package version(s)

Uno.WinUI 4.7.37, 4.9.26
SkiaSharp.Views.Uno.WinUI 2.88.3

**Comment by agneszitte** [comment]
@roubachof tested and said it was the same using latest Uno **5.0.0-dev.1380**

**Comment by carldebilly** [comment]
@roubachof On Android, I'm having far worse result...

![image](https://github.com/unoplatform/uno/assets/4174207/3207aa2f-c698-4ccb-83cd-51d1aaa9daa9)

**Comment by roubachof** [comment]
Yup this is another issue: a regression on layout with uno 5, it also impact ios.
You can either try it with version < 5, or toggle the theming to force a new layout pass and have the correct result (with bad padding :).

**Comment by eriklimakc** [comment]
On WASM there is clipping as well, but it looks different from Android. Could it be related?

![image](https://github.com/unoplatform/uno/assets/114886335/19fb2d51-54dc-4f2d-9394-82134cb2e369)

**Comment by eriklimakc** [comment]
About the WASM clipping, on SkiaSharp 2.88.4-preview.84 we get a better result:

![image](https://github.com/unoplatform/uno.toolkit.ui/assets/114886335/103312d7-6387-4f0e-97bb-87230c6d4d1a)

**Comment by jeromelaban** [comment]
There's still a little bit of clipping, but it's a lot better for sure.

**Comment by roubachof** [comment]
This is in fact the same clipping, but since the fix I created for wasm was integrated in 2.88.4-preview (https://github.com/mono/SkiaSharp/pull/2443), you have now the correct shadow color, which is softer and doesn't enlighten the wasm clipping so much.
